### PR TITLE
Add UMAT RVE-type example 

### PR DESCRIPTION
**Pull request by recuero**
Add demonstration example of the UMAT multiblock capability

Refs. #19249

### CODE REVIEW

File: modules/tensor_mechanics/test/tests/umat/multiple_blocks/tests [hunk 1]
```diff
@@ -31,5 +31,30 @@
...
                   'different blocks using distinct interface plug-ins in multi processor runs.'
   []
-
+  [rve_multimaterial_umat]
+    type = 'Exodiff'
+    input = 'rve_multimaterial.i'
+    exodiff = 'rve_multimaterial_umat_out.e'
+    cli_args = "Materials/active='umat_1 umat_2' "
+               "Outputs/file_base=rve_multimaterial_umat_out"
+    library_mode = 'DYNAMIC'
```
Comment (dschwen) [inline modules/tensor_mechanics/test/tests/umat/multiple_blocks/tests:40]: Need to add this to both tests:

```suggestion
    library_mode = 'DYNAMIC'
```

Comment (recuero) [reply modules/tensor_mechanics/test/tests/umat/multiple_blocks/tests:40]: Why to those two tests?

Comment (recuero) [reply modules/tensor_mechanics/test/tests/umat/multiple_blocks/tests:40]: Okay, I am reverting those new options since I am getting parse errors for now.

Comment (recuero) [reply modules/tensor_mechanics/test/tests/umat/multiple_blocks/tests:40]: It appears it's only needed in the combined module, per @milljm's PR https://github.com/milljm/moose/blob/b66fa8b937baf04917185244806d51a6d84b335b/modules/combined/test/tests/umat/tests#L23

**Review by dschwen — APPROVED** [review]
:+1:

**Review by dschwen — APPROVED** [review]
:+1:

**Pull request merged**